apiserver/facades/controller/crosscontroller/crosscontroller.go

```diff
@@ -32,7 +33,18 @@ func NewStateCrossControllerAPI(ctx facade.Context) (*CrossControllerAPI, error)
 	st := ctx.State()
 	return NewCrossControllerAPI(
 		ctx.Resources(),
-		func() ([]string, string, error) { return common.StateControllerInfo(st) },
+		func() ([]string, string, error) {
+			addrs, caCert, err := common.StateControllerInfo(st)
+			config, err := st.ControllerConfig()
+			if err != nil {
+				return nil, "", errors.Trace(err)
+			}
+			publicnDNSaddress := config.PublicDNSAddress()
+			if publicnDNSaddress != "" {
+				addrs = append(addrs, publicnDNSaddress)
```

Comment:
Should put first in the slice of addresses

Reply:
done